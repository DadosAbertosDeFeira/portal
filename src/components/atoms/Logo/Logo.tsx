import { Image } from "atoms/Image";
import type { LinkProps } from "atoms/Link";
import { Link } from "atoms/Link";
import classNames from "classnames";
import React from "react";

import DadosAbertosDeFeira from "../../../../public/assets/logo.svg";

export type LogoProps = Partial<LinkProps> & {};

export function Logo({ className, ...props }: LogoProps) {
  return (
    <Link
      href="/"
      className={classNames(" w-full max-w-[120px] h-full relative", className)}
      {...props}
    >
      <Image fill src={DadosAbertosDeFeira} alt="Logo Dados Abertos de Feira" />
    </Link>
  );
}
