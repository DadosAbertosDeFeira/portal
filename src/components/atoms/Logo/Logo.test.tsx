import { render } from "@testing-library/react";

import { Logo, type LogoProps } from "./Logo";

describe("Logo", () => {
  const makeSut = ({ ...props }: Partial<LogoProps>) => {
    return render(<Logo {...props} />);
  };

  it("renders correctly", () => {
    const { asFragment } = makeSut({});

    expect(asFragment).toMatchSnapshot();
  });

  it("renders with sizes correctly", () => {
    const { asFragment } = makeSut({ width: 40, height: 40 });

    expect(asFragment).toMatchSnapshot();
  });

  it("should render small version correctly", () => {
    const { asFragment } = makeSut({ small: true });

    expect(asFragment).toMatchSnapshot();
  });
});
