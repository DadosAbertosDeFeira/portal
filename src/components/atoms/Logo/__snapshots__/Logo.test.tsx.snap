// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `[Function]`;

exports[`Logo renders with sizes correctly 1`] = `[Function]`;

exports[`Logo should render small version correctly 1`] = `[Function]`;
